firmware/projects/DemoProjectRTOS/main.cc

```diff
@@ -0,0 +1,115 @@
+/// @author Blake Freer, Matteo Tullo
+/// @date 2023-12-25
+
+#include <cstdint>
+#include <cstring>
+
+#include "app.h"
+
+namespace bindings {
+extern shared::periph::DigitalInput& button_di;
+extern shared::periph::DigitalOutput& indicator_do;
+extern shared::periph::DigitalOutput& indicator2_do;
+extern void Initialize();
+}   // namespace bindings
+
+namespace os {
+extern shared::os::Semaphore& sem_test;
+extern shared::os::Fifo& message_queue_test;
+extern shared::os::Timer& timer_test;
+extern void Tick(uint32_t ticks);
+extern void TickUntil(uint32_t ticks);
+extern uint32_t GetTickCount();
+extern void InitializeKernel();
+extern void StartKernel();
+}   // namespace os
+
+extern "C" {
+void StartTestTask(void *argument);
```

Comment:
specify function arguments so the person writing the platform logic knows what to put here

Reply:
nevermind, I see that these are used for passing arbitrary items into tasks